lib/usd/utils/MayaTransformAPI.cpp

```diff
@@ -0,0 +1,1046 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include "mayaUsdUtils/MayaTransformAPI.h"
+
+namespace MayaUsdUtils {
+
+namespace {
+namespace tokens {
+const TfToken rotateXYZ("xformOp:rotateXYZ");
+const TfToken rotateXZY("xformOp:rotateXZY");
+const TfToken rotateYXZ("xformOp:rotateYXZ");
+const TfToken rotateYZX("xformOp:rotateYZX");
+const TfToken rotateZXY("xformOp:rotateZXY");
+const TfToken rotateZYX("xformOp:rotateZYX");
+const TfToken rotateAxis("xformOp:rotateXYZ:rotateAxis");
+const TfToken scale("xformOp:scale");
+const TfToken translate("xformOp:translate");
+const TfToken pivot("xformOp:translate:pivot");
+const TfToken pivotINV("!invert!xformOp:translate:pivot");
+const TfToken rotatePivot("xformOp:translate:rotatePivot");
+const TfToken scalePivot("xformOp:translate:scalePivot");
+const TfToken rotatePivotINV("!invert!xformOp:translate:rotatePivot");
+const TfToken scalePivotINV("!invert!xformOp:translate:scalePivot");
+const TfToken rotatePivotTranslate("xformOp:translate:rotatePivotTranslate");
+const TfToken scalePivotTranslate("xformOp:translate:scalePivotTranslate");
+const TfToken shear("xformOp:transform:shear");
+
+const TfToken scalePivotName("scalePivot");
+const TfToken rotatePivotName("rotatePivot");
+const TfToken scalePivotTranslateName("scalePivotTranslate");
+const TfToken rotatePivotTranslateName("rotatePivotTranslate");
+const TfToken rotateAxisName("rotateAxis");
+const TfToken shearName("shear");
+
+#if AL_SUPPORT_LEGACY_NAMES
+const TfToken rotateXYZOld("xformOp:rotateXYZ:rotate");
+const TfToken rotateXZYOld("xformOp:rotateXZY:rotate");
+const TfToken rotateYXZOld("xformOp:rotateYXZ:rotate");
+const TfToken rotateYZXOld("xformOp:rotateYZX:rotate");
+const TfToken rotateZXYOld("xformOp:rotateZXY:rotate");
+const TfToken rotateZYXOld("xformOp:rotateZYX:rotate");
+const TfToken translateOld("xformOp:translate:translate");
+const TfToken rotatePivotINVOld("xformOp:translate:rotatePivotINV");
+const TfToken scalePivotINVOld("xformOp:translate:scalePivotINV");
+const TfToken scaleOld("xformOp:scale:scale");
+#define XFORM_NAME_CHECK(newName) (name == (newName) || name == (newName ## Old))
+#else
+#define XFORM_NAME_CHECK(newName) (name == (newName))
+#endif
+}
+}
+
+//----------------------------------------------------------------------------------------------------------------------
+bool MayaTransformAPI::matchesMayaTrasformProfile(const std::vector<UsdGeomXformOp>& orderedOps)
```

Comment:
Awkward that this isn't const.  Could m_api be mutable, or does this pose a problem for object invariants and semantics?